core/src/main/java/software/amazon/awssdk/runtime/StandardMemberCopier.java

```diff
@@ -74,6 +74,19 @@ public static ByteBuffer copy(ByteBuffer bb) {
         if (bb == null) {
             return null;
         }
-        return bb.duplicate();
+
+        ByteBuffer copy = bb.isDirect() ? ByteBuffer.allocateDirect(bb.capacity()) : ByteBuffer.allocate(bb.capacity());
+
+        // Copy content
+        ByteBuffer toCopy = bb.asReadOnlyBuffer();
+        toCopy.rewind();
```

Comment:
There's no way to avoid the copying, since ByteBuffers can be backed by an array, and an array is inherently mutable. I'll copy the remaining data and update the javadoc to make it clear it's not a full copy.